src/sphinx.h

```diff
@@ -332,10 +332,22 @@ std::unique_ptr<ISphFieldFilter> sphCreateFilterICU ( std::unique_ptr<ISphFieldF
 // SEARCH QUERIES
 /////////////////////////////////////////////////////////////////////////////
 
+enum class EStrCmpDir:BYTE
```

Comment:
```suggestion
enum class EStrCmpDir
```
(Let's compiler decide, which underneath type is better. Optimize into 1 byte may be arguable when you have millions of similar filters, which is usually not the case. On the other side, default mem template most probably will anyway allocate 32-bits, or even 64-bits word, which will include low byte of your enum, and then 'padding' with undefined values. If you, say, want to cache queries, and reveal equal by hashing the mem region of the structure - that padding bytes would produce false misses in the hash. So, default enum size is quite enough in most of the cases, let us live it as is.

Reply:
Shouldn't we declare it as a `BYTE` because it must be a byte for the following line (`src/searchd.cpp:1816`)?

```
tOut.SendByte ( (BYTE)tFilter.m_eStrCmpDir );
```

It compiles fine as just `enum class EStrCmpDir ` but if the compiler ever optimizes it to not be a byte, wouldn't it lead to a silent problem?

Reply:
If once we reach so many comparision modes to not fit in single byte - we just raise proto version and use something else. Not sure, it will happen in our life, however ).
Problem between int and byte is that both occupy 4 bytes in the class. But int has all bytes known (i.e. 0), and byte has 3 bytes of padding which are not known.